x-pack/legacy/plugins/security/public/views/management/roles_grid/components/roles_grid_page.tsx

```diff
@@ -218,6 +220,62 @@ class RolesGridPageUI extends Component<Props, State> {
           ) : null;
         },
       },
+      {
+        name: intl.formatMessage({
+          id: 'xpack.security.management.roles.actionsColumnName',
+          defaultMessage: 'Actions',
+        }),
+        width: '150px',
+        actions: [
+          {
```

Comment:
I'll defer to others for their opinion on the addition of the "edit" button. I'd be tempted to just leave-off the dedicated edit button and only add the "clone" button. 

I find it somewhat awkward that for reserved roles, when you click the role-name it takes you to the edit form in a read-only view and there is no edit button. Where-as, for non-reserved roles, when you click the role-name it takes you to the edit view and there is an edit button which takes you to the same form.

Reply:
I took this route because it's consistent with the spaces grid page:
![image](https://user-images.githubusercontent.com/3493255/59884533-a7b3f400-9386-11e9-8ba6-f693de8027fc.png)


I do agree that the current implementation is awkward with reserved roles though. How about I give a "view" action for those, rather than an "edit" action?

Reply:
> I do agree that the current implementation is awkward with reserved roles though. How about > I give a "view" action for those, rather than an "edit" action?

That sounds good to me.

Reply:
I ended up forgoing the "view" action, and I'm only showing the "edit" action for roles that aren't considered readonly